app/helpers/Auth0Config.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Lightbend
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package helpers
+
+import play.api.Play
+
+case class Auth0Config(secret: String, clientId: String, callbackURL: String, domain: String)
+object Auth0Config {
+  def get() = {
+    Auth0Config(
+      Play.current.configuration.getString("auth0.clientSecret").get,
```

Comment:
`Play.current` is deprecated. The best way to have access to `Configuration` is using [Dependency Injection](https://www.playframework.com/documentation/2.6.x/ScalaDependencyInjection). You can rewrite this to use a [`Provider`](https://github.com/google/guice/wiki/ProviderBindings):

```scala
class Auth0ConfigProvider @Inject()(configuration: Configuration) extends Provider[Auth0Config] {
  override def get = {
    Auth0Config(
      secret = configuration.get[String]("auth0.clientSecret"),
      clientId = configuration.get[String]("auth0.clientId"),
      callbackURL = configuration.get[String]("auth0.callbackURL"),
      domain = configuration.get[String]("auth0.domain")
    )
  }
}
```

And then have a module to declare the binding:

```scala
import play.api.inject._

class Auth0Module extends SimpleModule(
  bind[Auth0Config].toProvider[Auth0ConfigProvider]
)
```
And finally register the `Auth0Module` as described [here](https://playframework.com/documentation/2.6.x/ScalaDependencyInjection#Programmatic-bindings):

```HOCON
play.modules.enabled += "helpers.Auth0Module"
```
While this looks more complicated and verbose, it will make it easier to write tests, since it is more flexible.

Reply:
It won't be so verbose once you start grouping multiple things into one module. Also in Guice you can do:

```scala
class AuthModule extends AbstractModule {
  @Provides @Singleton def auth0Config(conf: Configuration) = Auth0Config(
    conf.get[String]("auth0.clientSecret"),
    conf.get[String]("auth0.clientId"),
    conf.get[String]("auth0.callbackUrl"),
    conf.get[String]("auth0.domain")
  )
}
```